lib/quirc.c

```diff
@@ -86,6 +89,22 @@ int quirc_resize(struct quirc *q, int w, int h)
 			goto fail;
 	}
 
+	/*
+	 * alloc the work area for the flood filling logic.
+	 *
+	 * the size was chosen with the following assumptions and observations:
+	 *
+	 * - rings are the regions which requires the biggest work area.
+	 * - they consumes the most when they are rotated by about 45 degree.
+	 *   in that case, the necessary depth is about (2 * height_of_the_ring).
+	 * - the maximum height of rings would be about 1/3 of the image height.
+	 */
+
+	num_vars = h * 2 / 3;
+	vars = malloc(sizeof(*vars) * num_vars);
```

Comment:
any reason to use malloc instead of calloc here?

Reply:
just because it doesn't need to be initialized.

Reply:
uninitialized is fine, I meant for the overflow check safety net.

Reply:
i added a few checks. is this what you meant?